Deepak_Yadav/tic tac toe/scripts.js

```diff
@@ -0,0 +1,170 @@
+    var b5 = document.getElementById("b5");
+    var b6 = document.getElementById("b6");
+    var b7 = document.getElementById("b7");
+    var b8 = document.getElementById("b8");
+    var b9 = document.getElementById("b9");
+//initializing buttons with # value
+    b1.innerHTML = "#";
+    b2.innerHTML = "#";
+    b3.innerHTML = "#";
+    b4.innerHTML = "#";
+    b5.innerHTML = "#";
+    b6.innerHTML = "#";
+    b7.innerHTML = "#";
+    b8.innerHTML = "#";
+    b9.innerHTML = "#";
+
+    b1.style.backgroundColor = "#99780b";
+    b2.style.backgroundColor = "#99780b";
+    b3.style.backgroundColor = "#99780b";
+    b4.style.backgroundColor = "#99780b";
+    b5.style.backgroundColor = "#99780b";
+    b6.style.backgroundColor = "#99780b";
+    b7.style.backgroundColor = "#99780b";
+    b8.style.backgroundColor = "#99780b";
+    b9.style.backgroundColor = "#99780b";
+
+    b1.disabled = false;
+    b2.disabled = false;
+    b3.disabled = false;
+    b4.disabled = false;
+    b5.disabled = false;
+    b6.disabled = false;
+    b7.disabled = false;
+    b8.disabled = false;
+    b9.disabled = false;
+}
+
+//Winning combination
+function checker(){
+    //storing values in variables
+    var b1 = document.getElementById("b1").innerHTML;
+    var b2 = document.getElementById("b2").innerHTML;
+    var b3 = document.getElementById("b3").innerHTML;
+    var b4 = document.getElementById("b4").innerHTML;
+    var b5 = document.getElementById("b5").innerHTML;
+    var b6 = document.getElementById("b6").innerHTML;
+    var b7 = document.getElementById("b7").innerHTML;
+    var b8 = document.getElementById("b8").innerHTML;
+    var b9 = document.getElementById("b9").innerHTML;
+    
+
+    if(((b1=="X") || (b1=="O")) && ((b1 == b2) && (b2 == b3))){
+        highLight("b1", "b2", "b3");
+    }
+    else if(((b1=="X") || (b1=="O")) && ((b1 == b4) && (b4 == b7))){
+        highLight("b1", "b4", "b7");
+    }
+    else if(((b1=="X") || (b1=="O")) && ((b1 == b5) && (b5 == b9))){
+        highLight("b1", "b5", "b9");
+    }
+    else if(((b2=="X") || (b2=="O")) && ((b2 == b5) && (b5 == b8))){
+        highLight("b2", "b5", "b8");
+    }
+    else if(((b3=="X") || (b3=="O")) && ((b3 == b6) && (b6 == b9))){
+        highLight("b3", "b6", "b9");
+    }
+    else if(((b3=="X") || (b3=="O")) && ((b3 == b5) && (b5 == b7))){
+        highLight("b3", "b5", "b7");
+    }
+    else if(((b4=="X") || (b4=="O")) && ((b4 == b5) && (b5 == b6))){
+        highLight("b4", "b5", "b6");
+    }
+    else if(((b7=="X") || (b7=="O")) && ((b7 == b8) && (b8 == b9))){
+        highLight("b7", "b8", "b9");
+    }
+}
+
+
+//Disables the buttons, and paints the button's ORANGE, which are in WINNING pattern.
+//It takes buttons id's as the three arguments and symbol=X or O as another argument, but the symbol īs currently not in use.
+
+//The button pressed will be disabled and symbol 'X' or 'O' will appear on it
+function setButton(button, options){
+    if(button == 1){
```

Comment:
missing DRY, this function can be code with 5 to 6 lines only, rest is repeated code